### PR TITLE
Rename symbols for a more consistent use

### PR DESCRIPTION
**Pull request by mxz42**
- an interface (`IServerModule`) stays the same
- the dyn new type is prefixed with `Dyn` (`DynIServerModule`)
- the typed "plugin" (`ServerModulePlugin`) is now just `ServerModule` that enables implementors to write `impl ServerModule for ...`

I also renamed the `struct VerificationCache` to `struct VerifiedNotes` as it conflicted with the typed trait `VerificationCache`. 

part of #1232

### CODE REVIEW

File: client/client-lib/src/api.rs [hunk 1]
```diff
@@ -100,10 +100,10 @@ pub trait IFederationApi: Debug + Send + Sync {
 }
 
 dyn_newtype_define! {
-    pub FederationApi(Arc<IFederationApi>)
+    pub DynIFederationApi(Arc<IFederationApi>)
```
Comment (dpc) [inline client/client-lib/src/api.rs:103]: I thought we are going to do `DynFederationApi` (here and elsewhere). The `I` seems needless. I'm already a bit sad about that `Dyn`. ;)

**Comment by dpc** [comment]
I would drop that `I`, but will live with it one way or the other.

For @elsirion to decide.

File: client/client-lib/src/api.rs (hunk 1 above)
Comment (mxz42) [reply client/client-lib/src/api.rs:103]: oh okay  :) well, at least it's not a very long prefix

**Comment by mxz42** [comment]
the fixup commit drops the `I` in `Dyn`. Does github offer some autosquash functionality when merging? otherwise I'd like to squash that myself before it's being merged.

also a test for  #1253  :)

**Comment by elsirion** [comment]
I'm sorry, but I think #1269 just sent you into rebase hell :grimacing: might be easier to just re-do the renames as long as you remember which types you renamed.

**Review by dpc — APPROVED** [review]

**Pull request merged**